common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -350,9 +350,10 @@ private int indexOf(int key) {
     /**
      * Returns the hashed index for the given key.
      */
-    private int hashIndex(int key) {
+    private int hashIndex(@k@ key) {
         // Allowing for negative keys by adding the length after the first mod operation.
-        return (key % keys.length + keys.length) % keys.length;
+        int hashCode = @HASH_CODE@;
```

Comment:
@nmittler - I'm not questioning the use of the `HASH_CODE` transformation helper but was recommending doing the following:
```java
return (@HASH_CODE@ % keys.length + keys.length) % keys.length;
```

I think the `HASH_CODE` transformation is useful to map from 64 to 32 bits.